SystemManager/Src/SM.cpp

```diff
@@ -122,14 +123,51 @@ void SystemManager::AMOperationTask(void* pvParameters)
 //     }
 // }
 
-// void SystemManager::PMOperationTask(void *pvParameters)
-// {
-//     TickType_t xNextWakeTime;
-//     xNextWakeTime = xTaskGetTickCount();
-//     while (true) {
-//         PM_instance.execute();
-//         vTaskDelayUntil(&xNextWakeTime, SM::PM_PERIOD_MS);
-//     }
-// }
+void SystemManager::PMOperationTask(void *pvParameters)
+{
+    PM::PathManager* path_manager = (PM::PathManager*)pvParameters;
+
+    const PM::SM_PM_Commands* pm_instructions;
+
+    TickType_t xNextWakeTime;
+    xNextWakeTime = xTaskGetTickCount();
+    while (true) {
+         // Read MessageQ from SM
+        uint8_t* msg_priority = 0;
+        void* message_pointer = NULL;
+        // Arbitrary 3ms timeout as no message is no problem, gives AM time to run
+        osMessageQueueGet(path_manager->getSmPmQueue(), message_pointer, msg_priority, 3);
```

Comment:
True, the NULL msg pointer might cause trouble I am not sure offhand.
If the call times out then it might be a good idea to have a flag in the pm_instructions saying that the message is not a new message. PM should still run.